doc/source/whatsnew/v1.0.0.rst

```diff
@@ -84,6 +84,7 @@ Performance improvements
 Bug fixes
 ~~~~~~~~~
 
+- Bug in :meth:`DataFrame.apply` that caused incorrect output with empty :class:`DataFrame` (:issue:`28202`, :issue:`21959`)
```

Comment:
this is specific to ``np.*`` and ``.nunique()``? if so can you be more specific

Reply:
I _think_ it'd apply to any reduction whose output isn't empty when the input is, should I say something to that effect?